### ISSUE TITLE
Reparenting a node with NodePath by drag and drop causes endless loop

### ISSUE THREAD

**Issue by easy82** [godotengine/godot#35014]
**Godot version:**
3.2 beta

**OS/device including version:**
Windows 10

**Issue description:**
Reparenting a child node by dragging and dropping onto a parent node causes an endless loop if the child node has an assigned NodePath property.

From the Command Prompt:
```
ERROR: Attempt to disconnect [SceneTree:1163] signal 'node_removed' while in emission callback '_node_removed' (in target [Object:22280]). Use CONNECT_DEFERRED (to be able to safely disconnect) or CONNECT_ONESHOT (for automatic disconnection) as connection flags.
   At: core/object.cpp:1522
ERROR: Attempt to disconnect [SceneTree:1163] signal 'node_removed' while in emission callback '_node_removed' (in target [Object:22280]). Use CONNECT_DEFERRED (to be able to safely disconnect) or CONNECT_ONESHOT (for automatic disconnection) as connection flags.
   At: core/object.cpp:1522
ERROR: Attempt to disconnect [SceneTree:1163] signal 'node_removed' while in emission callback '_node_removed' (in target [Object:22280]). Use CONNECT_DEFERRED (to be able to safely disconnect) or CONNECT_ONESHOT (for automatic disconnection) as connection flags.
   At: core/object.cpp:1522
```

Doing the same in 3.1.2 does not cause endless loop:
```
ERROR: emit_signal: Condition ' !target ' is true. Continuing..:
   At: core/object.cpp:1199
```

The issue is not reproducable in 3.0.6.

**Steps to reproduce:**
1) Create a node called "Player" with the following code and save it as a scene:
```
extends Node2D
export(NodePath) var root_node: NodePath = NodePath("..")
```
2) Create another node called "World" and add the "Player" to it as a child.
3) Change "Root Node" property to "World".
4) Dragging "Player" onto "World" cases endless loop.

**Minimal reproduction project:**
[NodePathDragDropCrash.zip](https://github.com/godotengine/godot/files/4049402/NodePathDragDropCrash.zip)

**Comment by MightyPrinny** [comment]
I've had this happen to me too using the focus neighbor properties on control nodes

**Comment by akien-mga** [comment]
Fixed by #35088.
